File: apps/ios/Wordocious/Sources/MoreGamesBand.swift
import SwiftUI
import WordociousCore

/// The More Games band (founder + JP, 2026-09-26): a full-width tile directly
/// UNDER the game grid — indigo accent, the ten small game icons in catalog
/// order, "N of 10 played", chevron — so nobody hunts for the extra games but
/// the page still opens on the Daily Challenge and the eight word games.
///
/// It is also the More Games "hero": every More Games daily played → filled
/// indigo, "MORE GAMES SWEEP!"; every one won → the gold Flawless treatment,
/// "FLAWLESS MORE GAMES!". Purely visual — derived from today's completions,
/// never a bonus row, XP or a leaderboard. Tap opens the sheet; Share (sweep
/// states only) shares the More Games card. Mirrors web more-games-band.tsx.
struct MoreGamesBand: View {
    let mode: HomeMode
    /// The More Games titles visible to this viewer (catalog ∩ remote flags).
    let modes: [HomeMode]
    let playMode: PlayMode
    let byMode: [String: DailyCompletion]
    let onOpen: () -> Void
    let onShare: () -> Void

    private let indigo = Color(hex: 0x4F46E5)

    var body: some View {
        let daily = moreDailyModes(modes)
        let tier: MoreSweepTier? = playMode == .daily ? moreSweepTier(byMode: byMode, modes: modes) : nil
        let gold = tier == .flawless
        let totals = moreTotals(byMode: byMode, modes: modes)
        let titleC: Color = tier == nil ? Theme.textPrimary : (gold ? Color(hex: 0x92400E) : .white)
        let subC: Color = tier == nil ? Theme.textMuted : (gold ? Color(hex: 0xB45309) : Color(hex: 0xE0E7FF))
        let subtitle: String = {
            if let t = tier, t == tier {
                let mins = Int(totals.totalTimeSeconds) / 60, secs = Int(totals.totalTimeSeconds) % 60
                return "All \(totals.total) \(gold ? "won" : "played") · \(mins):\(String(format: "%02d", secs)) · \(formatScore(totals.totalScore)) pts"
            }
            return playMode == .daily ? morePlayedText(completedKeys: Set(byMode.keys), modes: modes) : mode.desc
        }()

        ZStack(alignment: .topTrailing) {
            Button(action: onOpen) {
                HStack(spacing: 12) {
                    ZStack {
                        RoundedRectangle(cornerRadius: 8).fill(tier == nil ? indigo.opacity(0.08) : Color.white.opacity(0.25))
                        if gold {
                            Image(systemName: "trophy.fill").font(.system(size: 18)).foregroundStyle(Color(hex: 0xB45309))
                        } else if tier != nil {
                            Image(systemName: "sparkles").font(.system(size: 18)).foregroundStyle(.white)
                        } else {
                            Image(systemName: "square.grid.2x2").font(.system(size: 18, weight: .bold)).foregroundStyle(indigo)
                        }
                    }
                    .frame(width: 36, height: 36)
                    VStack(alignment: .leading, spacing: 3) {
                        Text(tier?.title ?? mode.title).font(Brand.font(13, .black)).foregroundStyle(titleC).lineLimit(1)
                        // The ten game icons, catalog order; in a sweep state every tile is a check.
                        HStack(spacing: 4) {
                            ForEach(daily) { m in
                                let done = m.dbKey.map { byMode[$0] != nil } ?? false
                                if tier != nil {
                                    ZStack {
                                        RoundedRectangle(cornerRadius: 5).fill(Color.white.opacity(0.9))
                                        Image(systemName: "checkmark").font(.system(size: 10, weight: .black))
                                            .foregroundStyle(gold ? Color(hex: 0xB45309) : indigo)
                                    }
                                    .frame(width: 18, height: 18)
                                } else if done {
                                    // Played today = a SOLID accent chip with a white glyph (founder, 2026-09-26:
                                    // "I finished Spyglass and you can barely tell" — the ring alone was too quiet).
                                    ZStack {
                                        RoundedRectangle(cornerRadius: 5).fill(m.accent)
                                        ModeIconView(icon: m.icon, accent: .white, box: 18)
                                    }
                                    .frame(width: 18, height: 18)
                                    .shadow(color: m.accent.opacity(0.45), radius: 3, y: 1)
                                } else {
                                    // Still to play = faded tint.
                                    ModeIconView(icon: m.icon, accent: m.accent, box: 18)
                                        .overlay(RoundedRectangle(cornerRadius: 5).stroke(m.accent.opacity(0.25), lineWidth: 1))
                                        .opacity(0.5)
                                }
                            }
                        }
                        .accessibilityHidden(true)
                        Text(subtitle).font(Brand.font(10, .bold)).foregroundStyle(subC).lineLimit(1)
                    }
                    Spacer(minLength: 4)
                    Image(systemName: "chevron.right").font(.system(size: 15, weight: .bold))
                        .foregroundStyle(tier == nil ? indigo : (gold ? Color(hex: 0xB45309) : .white))
                }
                .padding(.horizontal, 12).padding(.vertical, 10)
                .frame(maxWidth: .infinity)
                .background(
                    RoundedRectangle(cornerRadius: 14).fill(background(tier: tier, gold: gold))
                        .overlay(tier != nil ? BannerShimmer().clipShape(RoundedRectangle(cornerRadius: 14)) : nil)
                )
                .overlay(alignment: .leading) {
                    if tier == nil {
                        RoundedRectangle(cornerRadius: 14).fill(indigo).frame(width: 4).padding(.vertical, 6).padding(.leading, 0)
                    }
                }
                .overlay(RoundedRectangle(cornerRadius: 14).stroke(tier == nil ? Theme.border : (gold ? Color(hex: 0xF59E0B) : indigo), lineWidth: 1.5))
            }
            .buttonStyle(.plain)
            .accessibilityLabel(tier?.short ?? "More Games")
            .accessibilityHint(subtitle)

            if tier != nil {
                Button(action: onShare) {
                    Text("Share").font(Brand.font(10, .black))
                        .foregroundStyle(gold ? Color(hex: 0xB45309) : indigo)
                        .padding(.horizontal, 8).padding(.vertical, 3)
                        .background(Capsule().fill(Color.white.opacity(0.85)))
                }
                .buttonStyle(.plain)
                .padding(.top, 8).padding(.trailing, 36)
            }
        }
    }

    private func background(tier: MoreSweepTier?, gold: Bool) -> LinearGradient {
        if tier == nil { return LinearGradient(colors: [Theme.surface, Theme.surface], startPoint: .leading, endPoint: .trailing) }
        return gold
            ? LinearGradient(colors: [Color(hex: 0xFEF3C7), Color(hex: 0xFDE68A)], startPoint: .topLeading, endPoint: .bottomTrailing)
            : LinearGradient(colors: [indigo, Color(hex: 0x6366F1)], startPoint: .topLeading, endPoint: .bottomTrailing)
    }
}

File: apps/android/app/src/main/kotlin/com/wordocious/app/ui/MoreGamesBand.kt
package com.wordocious.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wordocious.app.data.DailyCompletionsService
import com.wordocious.app.ui.theme.Nunito
import com.wordocious.app.ui.theme.WTheme

private val INDIGO = Color(0xFF4F46E5)

/**
 * The More Games band (founder + JP, 2026-09-26): a full-width tile directly UNDER
 * the game grid — indigo accent, the ten small game icons in catalog order, "N of
 * 10 played", chevron — so nobody hunts for the extra games but the page still opens
 * on the Daily Challenge and the eight word games.
 *
 * It is also the More Games "hero": every More Games daily played → filled indigo,
 * "MORE GAMES SWEEP!"; every one won → the gold Flawless treatment, "FLAWLESS MORE
 * GAMES!". Purely visual — derived from today's completions, never a bonus row, XP
 * or a leaderboard. Tap opens the sheet; Share (sweep states only) shares the More
 * Games card. Mirrors web more-games-band.tsx / iOS MoreGamesBand.swift.
 */
@Composable
fun MoreGamesBand(
    card: ModeCard,
    /** The More Games titles visible to this viewer (catalog ∩ remote flags). */
    modes: List<ModeCard>,
    unlimitedMode: Boolean,
    completions: Map<String, DailyCompletionsService.Completion>,
    onOpen: () -> Unit,
    onShare: () -> Unit,
) {
    val daily = moreDailyModes(modes)
    val tier = if (unlimitedMode) null else moreSweepTier(completions, modes)
    val gold = tier == MoreSweepTier.FLAWLESS
    val totals = moreTotals(completions, modes)
    val titleC = if (tier == null) WTheme.text else if (gold) Color(0xFF92400E) else Color.White
    val subC = if (tier == null) WTheme.textMuted else if (gold) Color(0xFFB45309) else Color(0xFFE0E7FF)
    val subtitle = when {
        tier != null -> "All ${totals.total} ${if (gold) "won" else "played"} · ${totals.totalTimeSeconds / 60}:${"%02d".format(totals.totalTimeSeconds % 60)} · ${formatScore(totals.totalScore.toDouble())} pts"
        unlimitedMode -> card.desc
        else -> morePlayedText(completions.keys, modes)
    }
    val shape = RoundedCornerShape(14.dp)
    val background = when {
        tier == null -> Brush.linearGradient(listOf(WTheme.surface, WTheme.surface))
        gold -> Brush.linearGradient(listOf(Color(0xFFFEF3C7), Color(0xFFFDE68A)))
        else -> Brush.linearGradient(listOf(INDIGO, Color(0xFF6366F1)))
    }
    val borderC = when { tier == null -> WTheme.border; gold -> Color(0xFFF59E0B); else -> INDIGO }

    Box(Modifier.fillMaxWidth().clip(shape).background(background).border(1.5.dp, borderC, shape)) {
        if (tier == null) Box(Modifier.width(4.dp).fillMaxHeight().padding(vertical = 6.dp).clip(CircleShape).background(INDIGO))
        Row(
            Modifier.fillMaxWidth().clickableNoRipple(onOpen).padding(horizontal = 12.dp, vertical = 10.dp),
            verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            Box(
                Modifier.size(36.dp).clip(RoundedCornerShape(8.dp))
                    .background(if (tier == null) INDIGO.copy(alpha = 0.08f) else Color.White.copy(alpha = 0.25f)),
                contentAlignment = Alignment.Center,
            ) {
                when {
                    gold -> Icon(Icons.Filled.EmojiEvents, null, tint = Color(0xFFB45309), modifier = Modifier.size(20.dp))
                    tier != null -> Icon(Icons.Filled.AutoAwesome, null, tint = Color.White, modifier = Modifier.size(20.dp))
                    else -> Icon(Icons.Filled.GridView, null, tint = INDIGO, modifier = Modifier.size(20.dp))
                }
            }
            Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(3.dp)) {
                Text(tier?.title ?: card.title, fontSize = 13.sp, fontWeight = FontWeight.Black, color = titleC, maxLines = 1, overflow = TextOverflow.Ellipsis, fontFamily = Nunito)
                // The ten game icons, catalog order; in a sweep state every tile is a check.
                Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                    for (m in daily) {
                        val done = m.dbKey != null && completions.containsKey(m.dbKey)
                        if (tier != null) {
                            Box(Modifier.size(18.dp).clip(RoundedCornerShape(5.dp)).background(Color.White.copy(alpha = 0.9f)), contentAlignment = Alignment.Center) {
                                Icon(Icons.Filled.Check, null, tint = if (gold) Color(0xFFB45309) else INDIGO, modifier = Modifier.size(11.dp))
                            }
                        } else {
                            // Played today = full strength with an accent ring; still to play = faded.
                            Box(
                                Modifier.size(18.dp).clip(RoundedCornerShape(5.dp))
                                    .border(1.2.dp, m.accent.copy(alpha = if (done) 0.9f else 0.25f), RoundedCornerShape(5.dp))
                                    .alpha(if (done) 1f else 0.55f),
                                contentAlignment = Alignment.Center,
                            ) { ModeGlyph(m, m.accent, 18.dp) }
                        }
                    }
                }
                Text(subtitle, fontSize = 10.sp, fontWeight = FontWeight.Bold, color = subC, maxLines = 1, overflow = TextOverflow.Ellipsis)
            }
            Icon(Icons.Filled.ChevronRight, null, tint = if (tier == null) INDIGO else if (gold) Color(0xFFB45309) else Color.White, modifier = Modifier.size(20.dp))
        }
        if (tier != null) {
            Text(
                "Share", fontSize = 10.sp, fontWeight = FontWeight.Black, color = if (gold) Color(0xFFB45309) else INDIGO,
                modifier = Modifier.align(Alignment.TopEnd).padding(top = 8.dp, end = 36.dp)
                    .clip(CircleShape).background(Color.White.copy(alpha = 0.85f)).clickable { onShare() }
                    .padding(horizontal = 8.dp, vertical = 3.dp),
                style = TextStyle(fontFamily = Nunito),
            )
        }
    }
}
